### PR TITLE
ncm-network: support rules and routes using routing tables

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: ncm-network/src/main/pan/components/network/core-schema.pan [hunk 1]
```diff
@@ -328,6 +351,9 @@ type structure_ipv6 = {
     "gatewaydev" ? valid_interface # sets IPV6_DEFAULTDEV
 };
 
+type network_routing_table = {
+};
```
Comment (jrha) [inline ncm-network/src/main/pan/components/network/core-schema.pan]: Empty type?

Comment (stdweird) [reply ncm-network/src/main/pan/components/network/core-schema.pan]: fixed

**Comment by stdweird** [comment]
@jrha remark addressed

File: ncm-network/src/main/pan/components/network/core-schema.pan [hunk 2]
```diff
@@ -58,11 +64,28 @@ type structure_route = {
     Presence of ':' in any of the values indicates this is IPv6 related.
 }
 type structure_rule = {
+    @{to selector}
+    "to" ? network_ip_cmd_prefix
+    @{from selector}
+    "from" ? network_ip_cmd_prefix
+    @{not action (false value means no not action; also the default when not is not defined)}
```
Comment (jrha) [inline ncm-network/src/main/pan/components/network/core-schema.pan]: Uhm... I'm having a hard time parsing this... What does it do?

Comment (jrha) [reply ncm-network/src/main/pan/components/network/core-schema.pan]: ?

Comment (jrha) [reply ncm-network/src/main/pan/components/network/core-schema.pan]: @stdweird?

Comment (stdweird) [reply ncm-network/src/main/pan/components/network/core-schema.pan]: it's the inverse of a selector (not sure why i thought it was the inverse of an action) (`man ip rule`). i guess you want me to change it :wink:

Comment (jrha) [reply ncm-network/src/main/pan/components/network/core-schema.pan]: I want to understand it, which I guess means yes, I do want you to change it :stuck_out_tongue:

**Pull request merged**